R/zchunk_L222.en_transformation.R

```diff
@@ -81,248 +87,540 @@ module_energy_L222.en_transformation_DISABLED <- function(command, ...) {
+
+    # 2. Build tables for CSVs
+    # 2a. Supplysector information
+    # L222.Supplysector_en: Supply sector information for energy transformation sectors
+
+    A22.sector %>%
+      write_to_all_regions(c("region", "supplysector", "output.unit", "input.unit", "price.unit", "logit.year.fillout", "logit.exponent"), GCAM_region_names) ->
+      L222.Supplysector_en
+
+    # 2b. Subsector information
+    # L222.SubsectorLogit_en: Subsector logit exponents of energy transformation sectors
+
+    A22.subsector_logit %>%
+      write_to_all_regions(c("region", "supplysector", "subsector", "logit.year.fillout", "logit.exponent"), GCAM_region_names) ->
+      L222.SubsectorLogit_en
+
+    if (any(!is.na(A22.subsector_shrwt$year))){
+      A22.subsector_shrwt %>%
+        filter(!is.na(year)) %>%
+        write_to_all_regions(c("region", "supplysector", "subsector", "year", "share.weight"), GCAM_region_names) ->
+        L222.SubsectorShrwt_en
+    }
+
+    if (any(!is.na(A22.subsector_shrwt$year.fillout))){
+      A22.subsector_shrwt %>%
+        filter(!is.na(year.fillout)) %>%
+        write_to_all_regions(c("region", "supplysector", "subsector", "year.fillout", "share.weight"), GCAM_region_names) ->
+        L222.SubsectorShrwtFllt_en
+      }
+
+    # L222.SubsectorInterp_en and L222.SubsectorInterpTo_en: Subsector shareweight interpolation of energy transformation sectors
+
+    if (any(is.na(A22.subsector_interp$to.value))){
+    A22.subsector_interp %>%
+      filter(is.na(to.value)) %>%
+      write_to_all_regions(c("region", "supplysector", "subsector", "apply.to", "from.year", "to.year", "interpolation.function"), GCAM_region_names) ->
+      L222.SubsectorInterp_en
+    }
+
+    if (any(!is.na(A22.subsector_interp$to.value))){
+    A22.subsector_interp %>%
+      filter(!is.na(to.value)) %>%
+      write_to_all_regions(c("region", "supplysector", "subsector", "apply.to", "from.year", "to.year", "to.value", "interpolation.function"), GCAM_region_names) ->
+      L222.SubsectorInterpTo_en
+    }
+
+    # 2c. Technology information
+
+    # L222.StubTech_en: Identification of stub technologies of energy transformation
+    # set up filter to keep all non-first gen bio techs in L222.StubTech_en
+    firstgenbio_techs <- c("corn ethanol", "sugarbeet ethanol", "sugar cane ethanol", "biodiesel")
+
+    # create list of regional stub.technologies
+    A22.globaltech_shrwt %>%
+      write_to_all_regions(c("region", "supplysector", "subsector", "technology"), GCAM_region_names) %>%
+      rename(stub.technology = technology) %>%
+      # Drops region x technology combinations that are not applicable, i.e. any first gen bio techs not listed for a region in A_regions
+      filter(!(stub.technology %in% firstgenbio_techs) | paste(region, stub.technology) %in%
+               c(paste(A_regions$region, A_regions$ethanol), paste(A_regions$region, A_regions$biodiesel))) ->
```

Comment:
Could this be done with a `semi_join`? Seems like that would be a lot more elegant and clear.

Reply:
The best I've come up with is an anti_join followed by two semi_joins and then binding rows back together? Is there a way to use `semi_join` to replace `%in%` that allows you to use the boolean operators? Otherwise I'm creating 2 to 3 separate pipelines.

Reply:
Do whatever seems most sensible to you. Thanks.